readthedocs/rtd_tests/tests/test_doc_building.py

```diff
@@ -837,6 +839,229 @@ def test_command_oom_kill(self):
+
+        self.build_env_mock = Mock()
+
+    def test_install_core_requirements_sphinx(self):
+        python_env = Virtualenv(
+            version=self.version_sphinx,
+            build_env=self.build_env_mock,
+        )
+        python_env.install_core_requirements()
+        requirements = [
+            'Pygments==2.2.0',
+            'setuptools==37.0.0',
+            'docutils==0.13.1',
+            'mock==1.0.1',
+            'pillow==2.6.1',
+            'alabaster>=0.7,<0.8,!=0.7.5',
+            'commonmark==0.5.4',
+            'recommonmark==0.4.0',
+            'sphinx==1.6.5',
+            'sphinx-rtd-theme<0.3',
+            'readthedocs-sphinx-ext<0.6',
+        ]
+        args = [
+            'python',
+            mock.ANY,  # pip path
+            'install',
+            '--use-wheel',
+            '--upgrade',
+            '--cache-dir',
+            mock.ANY,  # cache path
+        ]
+        args.extend(requirements)
+        self.build_env_mock.run.assert_called_once_with(
+            *args, bin_path=mock.ANY
+        )
+
+    def test_install_core_requirements_mkdocs(self):
+        python_env = Virtualenv(
+            version=self.version_mkdocs,
+            build_env=self.build_env_mock
+        )
+        python_env.install_core_requirements()
+        requirements = [
+            'Pygments==2.2.0',
+            'setuptools==37.0.0',
+            'docutils==0.13.1',
+            'mock==1.0.1',
+            'pillow==2.6.1',
+            'alabaster>=0.7,<0.8,!=0.7.5',
+            'commonmark==0.5.4',
+            'recommonmark==0.4.0',
+            'mkdocs==0.15.0',
+        ]
+        args = [
+            'python',
+            mock.ANY,  # pip path
+            'install',
+            '--use-wheel',
+            '--upgrade',
+            '--cache-dir',
+            mock.ANY,  # cache path
+        ]
+        args.extend(requirements)
+        self.build_env_mock.run.assert_called_once_with(
+            *args, bin_path=mock.ANY
+        )
+
+    def test_install_user_requirements(self):
+        self.build_env_mock.project = self.project_sphinx
+        self.build_env_mock.version = self.version_sphinx
+        python_env = Virtualenv(
+            version=self.version_sphinx,
+            build_env=self.build_env_mock
+        )
+
+        checkout_path = python_env.checkout_path
+        docs_requirements = os.path.join(
+            checkout_path, 'docs', 'requirements.txt'
+        )
+        root_requirements = os.path.join(
+            checkout_path, 'requirements.txt'
+        )
+        paths = {
+            os.path.join(checkout_path, 'docs'): True,
+        }
+        args = [
+            'python',
+            mock.ANY,  # pip path
+            'install',
+            '--exists-action=w',
+            '--cache-dir',
+            mock.ANY,  # cache path
+            'requirements_file'
+        ]
+
+        paths[docs_requirements] = True
```

Comment:
Can you add a simple comment in each of the cases saying what's the scenario and what we expect?